### PR TITLE
Remove tests workaround for builds with accelerated hashes

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
I checked the places where we have PSA test cases or test code that depend on built-in support for the algorithm, conveyed by `MBEDTLS_PSA_BUILTIN_xxx` dependencies (I did not analyze the use of legacy symbols as dependencies in PSA tests). Following https://github.com/Mbed-TLS/mbedtls/pull/7432, the following uses of `MBEDTLS_PSA_BUILTIN_xxx` remain:

* Some derivation-of-ECC-key test cases in `test_suite_psa_crypto.data`. Justified. I added a comment.
* Some test cases in `test_suite_psa_crypto_entropy.data`. Justified, already explained in a comment.
* Some code in `test_suite_psa_crypto_storage_format.function` to skip exercising in certain cases, which should no longer be necessary since the md-light work.
* Some code related to testing the driver interface, in `test_suite_psa_crypto_driver_wrappers.data` and `tests/src/drivers/*`, which is obviously justified.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** no (test changes only)
- [x] **backport** no (limitations in 2.28 have not been lifted)
- [x] **tests** N/A (just changing test code)

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto_storage_format.function [hunk 1]
```diff
@@ -61,100 +61,6 @@ static int can_export(const psa_key_attributes_t *attributes)
     }
 }
 
-#if defined(MBEDTLS_TEST_LIBTESTDRIVER1)
```
Comment (gilles-peskine-arm) [inline tests/suites/test_suite_psa_crypto_storage_format.function:64]: Of note, this workaround did not require listing in the ignore list for `do_analyze_driver_vs_reference` in `analyze_outcomes.py`, because it only affects what certain test cases do and not which test cases run.

File: tests/suites/test_suite_psa_crypto_storage_format.function [hunk 2]
```diff
@@ -193,24 +122,11 @@ static int can_exercise(const psa_key_attributes_t *attributes)
         return 0;
     }
 #endif
+
     if (is_accelerated_rsa(alg) &&
         (hash_alg == PSA_ALG_RIPEMD160 || hash_alg == PSA_ALG_SHA_384)) {
         return 0;
     }
-#if defined(MBEDTLS_PSA_ACCEL_ALG_RSA_OAEP)
```
Comment (mpg) [inline tests/suites/test_suite_psa_crypto_storage_format.function:200]: Note: this change (that part about RSA OAEP) is not reflected in the commit message, but seems correct though.

**Review by mpg — DISMISSED** [review]
LGTM.

I have not double-checked that all instances of `MBEDTLS_PSA_BUILTIN_` that could be removed indeed were, but even if that weren't the case, this PR would already be a strict improvement over the existing.

File: tests/suites/test_suite_psa_crypto_storage_format.function [hunk 3]
```diff
@@ -193,24 +122,11 @@ static int can_exercise(const psa_key_attributes_t *attributes)
...
-    if (PSA_ALG_IS_RSA_OAEP(alg) &&
-        (hash_alg == PSA_ALG_RIPEMD160 || hash_alg == PSA_ALG_SHA_384)) {
-        return 0;
```
Comment (gowthamsk-arm) [inline tests/suites/test_suite_psa_crypto_storage_format.function:203]: nit: Can be moved to a separate commit.

**Review by gowthamsk-arm — APPROVED** [review]
LGTM.

**Comment by gilles-peskine-arm** [comment]
I pushed a merge commit to resolve the merge conflict.

File: tests/suites/test_suite_psa_crypto.data [hunk 4]
```diff
@@ -6629,6 +6629,13 @@ PSA key derivation: PBKDF2-AES-CMAC-PRF-128, exercise HMAC-SHA-256
 depends_on:PSA_WANT_ALG_PBKDF2_AES_CMAC_PRF_128:PSA_WANT_ALG_CMAC:PSA_WANT_ALG_HMAC:PSA_WANT_ALG_SHA_256:PSA_WANT_KEY_TYPE_AES:PSA_WANT_KEY_TYPE_HMAC
 derive_key_exercise:PSA_ALG_PBKDF2_AES_CMAC_PRF_128:"706173737764":"01":"73616c74":PSA_KEY_TYPE_HMAC:256:PSA_KEY_USAGE_SIGN_HASH:PSA_ALG_HMAC(PSA_ALG_SHA_256)
 
+# Deriving a key on a Weierstrass curve requires a driver call, but we
+# haven't implemented cooked key derivation drivers yet
+# ("key_derivation_output_key" driver entry point). Until then,
+# require built-in support for the curve of the derived key.
+# This limitation does not apply to keys on a Montgomery curve for which
+# derivation doesn't involve potentially-accelerated operations and thus
+# doesn't involve a driver call.
```
Comment (mpg) [inline tests/suites/test_suite_psa_crypto.data:6638]: I think you no longer need this, since #8075.

Requiring `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` should be enough to enable `ECP_LIGHT` and the built-in version of all required curves. See https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/config_adjust_legacy_from_psa.h#L187 then https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/config_adjust_legacy_from_psa.h#L207 and https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/config_adjust_legacy_from_psa.h#L443

**Review by mpg — CHANGES_REQUESTED** [review]
Sorry, noticed something I missed in my previous review.

**Comment by mpg** [comment]
I think the 2nd commit [is no longer needed](https://github.com/Mbed-TLS/mbedtls/pull/7941#discussion_r1379769852) and I've cherry-picked the first one in #8498 which needed it too. So, I think this PR can be closed as superseded by #8498 but I'll leave that up to you, @gilles-peskine-arm

**Comment by gilles-peskine-arm** [comment]
Superseded by #8498.
